csm/conf/setup.py

```diff
@@ -881,12 +891,15 @@ def post_update(self, args):
         try:
             Log.info(f"Triggering csm_setup post_update: {args}")
             if self._is_user_exist():
-                Log.debug(f"Deleting user {self._user}")
-                Setup._run_cmd(f"userdel -r {self._user}")
+                csm_passwd = self._fetch_csm_user_password(decrypt=True)
+                cmd = (f"bash -c \"echo -e '{csm_passwd}\\n{csm_passwd}' | passwd {self._user}\"")
```

Comment:
We should check for whether `csm_passwd ` is empty or not. What if due to some reason password is not generated or fetched from provisioner.

Reply:
Comment Address